lib/browser-auto-tracking.js

```diff
@@ -68,7 +69,10 @@ export function initAutoTrackingCore(lib) {
       }
     }
 
```

Comment:
@jeremygarey -- I think this logic could be handled more cleanly by a single outer check for `if (!options.disableCookies) {` starting above the cookie instantiation and closing below the `initialReferrer` logic.

Then you can eliminate a bunch of standalone checks of the `options.disableCookies` flag.

Reply:
@EyePulp - Yeah definitely, I was thinking about that because I don't love the logic as it stands. But doing one if statement would completely remove `uuid` and `initialReferrer` logic altogether if cookies are disabled. Is that something we'd be okay with?

Reply:
@jeremy - great point, you're correct.

Wrapping it all in one `if` is super clean, but has the drawbacks you mention.

Other options:

## Monkey Patching
Since instantiating a cookie ( `cookie = new utils.cookie('keen');`) doesn't  actually create a cookie --
We could just no-op the the `.set` function...

```js 
    let cookie;
    cookie = new utils.cookie('keen);
    if (options.disableCookies) {
      cookie.set = () => { /* haha, no cookies for you */  };
    }
```
And then the rest of the code is untouched.  This is called monkey-patching and is probably not the right solution.  Maybe it is time we branched and started cleaning this up for our purposes.

## Only prevent `.set()`

Just wrap `cookie.set()` calls with `if (!options.disableCookies) {` -- `cookie.get()` doesn't break if there is no cookie, and does not create a missing cookie.  It appears only `.set()` creates a cookie.

## Wrap every cookie related call

This is where  the PR is now.  I expect it works, but it's more logic than necessary.

Reply:
@EyePulp - created a new [PR](https://github.com/mirusresearch/keen-tracking.js/pull/2) with changes since this one has already been merged.